### PR TITLE
static: reinstate fixed navbar via flex

### PR DESCRIPTION
**Pull request by haraldschilly**
# Description

# Testing Steps
I tested this in chrome and FF

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):
- [ ] No debugging console.log messages.
- [ ] All new code is actually used.
- [ ] Non-obvious code has some sort of comments.

Front end:
- [ ] Restart at least one project and check its `~/.smc/local_hub/local_hub.log`
- [ ] Completely restart Webpack with `./w` in `/src`
- [ ] Completely restart the hub by killing and restarting `./start_hub.py` in `/src/dev/project`
- [ ] Screenshots if relevant.

### CODE REVIEW

**Comment by williamstein** [comment]
What did you change?  It's a LOT of work to tell from this patch, because you moved all the relevant code to another file, then changed it there...

**Comment by williamstein** [comment]
Well, I don't understand this, but it seems harmless, and worse case I can revert.

**Comment by haraldschilly** [comment]
yes, I also cleaned this up a bit, makes it hard to follow. The key idea is to split the top part in two divs and then

```
body
  display: flex
  flex-direction: column
  height: 100vh
  overflow: hidden

div#content
  flex: 1 1 auto
  overflow: auto

nav#nav-main
  flex: 0 0 auto
```
